commands/build.go

```diff
@@ -412,12 +416,40 @@ func runBuild(ctx context.Context, dockerCli command.Cli, options buildOptions)
 }
 
 // getImageID returns the image ID - the digest of the image config
-func getImageID(resp map[string]string) string {
-	dgst := resp[exptypes.ExporterImageDigestKey]
-	if v, ok := resp[exptypes.ExporterImageConfigDigestKey]; ok {
-		dgst = v
+func getImageID(resp map[string]string, dockerUsingContainerdSnapshotter bool) string {
+	if dockerUsingContainerdSnapshotter {
+		// If Docker is using the containerd snapshotter, use the image digest, not
+		// the image config digest. See https://github.com/moby/moby/issues/45458.
+		return resp[exptypes.ExporterImageDigestKey]
+	}
+
+	return resp[exptypes.ExporterImageConfigDigestKey]
```

Comment:
This is not compatible with previous code there image digest is returned if config digest does not exist.

Reply:
Thanks, I've changed it to be compatible with how it was (when `preferImageDigest` is false).